### PR TITLE
Added a new image classification exercise

### PR DESCRIPTION
**Pull request by ameliepeter**
## Description

Added a new image classification exercise involving pre-trained models. 

Fixes # (issue)

## Tested on 
- [X] Windows  
- [ ] Linux Ubuntu  
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

Assign a reviewer to review your code.

### CODE REVIEW

**Comment by guanyu-certifai** [comment]
Great job on developing the module @ameliepeter! I feel it would be slightly tidier if these three models could be wrapped up in their own respective function, and pass in hyperparameters as arguments into them.

**Comment by guanyu-certifai** [comment]
Review done @kenghooi-teoh pretty clear explanations and comments

**Review by KianYang-Lee — APPROVED** [review]
Reviewed and find it is good to merge @kenghooi-teoh

**Comment by kenghooi-teoh** [comment]
@KianYang-Lee here's a few changes that I suggest to improve the explanation of tasks and parts of the code:

1. _Change the exercise instructions to the following:_

> _In this exercise, we will try to do a simple hands-on exercise to assess the relationship between CNN's complexity and the achieved model performance for a given dataset._
> _You may use a different image dataset than the 4-class Weather Image dataset provided (Cloudy/Rain/Shine/Sunrise) in the WeatherDataSetIterator._
> 
> _To make your comparison meaningful:_
> - _Try models with different architectures e.g. VGG16, VGG19, SqueezeNet with noticeable difference in terms of model size._ 
> - _Train each of the 3 models on a common set of hyperparameters e.g. learning rate, updater, batch size etc. and image preprocessing methods as well as the dataset splits._
> - _Keep a record of evaluation metrics among these 3 models and compare their performance. The comparison should highlight how the differences in  model architectures (e.g. number of layers, use of skip connections, use of Batch-norm etc.) affect the achieved scores, and you may also include other points of discussion that reasonably explains the disparity/similarity of model performance for the given dataset._

2. In line 33, Amelie mentioned that dataset is stored in "resources" folder, but the iterator file downloads the dataset into the data folder in home directory instead of resources folder.

3. To make the code a little cleaner in the Main class, you may wrap the 3 chunks of Model configuration code into a getModelConfig() method

4. An example of comparison can be provided in the solution folder, as a txt file or other file formats that is readable

**Comment by KianYang-Lee** [comment]
> @KianYang-Lee here's a few changes that I suggest to improve the explanation of tasks and parts of the code:
> 
> 1. Change the exercise instructions to the following:
> 
> > _In this exercise, we will try to do a simple hands-on exercise to assess the relationship between CNN's complexity and the achieved model performance for a given dataset._
> > _You may use a different image dataset than the 4-class Weather Image dataset provided (Cloudy/Rain/Shine/Sunrise) in the WeatherDataSetIterator._
> > _To make your comparison meaningful:_
> > 
> > * _Try models with different architectures e.g. VGG16, VGG19, SqueezeNet with noticeable difference in terms of model size._
> > * _Train each of the 3 models on a common set of hyperparameters e.g. learning rate, updater, batch size etc. and image preprocessing methods as well as the dataset splits._
> > * _Keep a record of evaluation metrics among these 3 models and compare their performance. The comparison should highlight how the differences in  model architectures (e.g. number of layers, use of skip connections, use of Batch-norm etc.) affect the achieved scores, and you may also include other points of discussion that reasonably explains the disparity/similarity of model performance for the given dataset._
> 
> 1. In line 33, Amelie mentioned that dataset is stored in "resources" folder, but the iterator file downloads the dataset into the data folder in home directory instead of resources folder.
> 2. To make the code a little cleaner in the Main class, you may wrap the 3 chunks of Model configuration code into a getModelConfig() method
> 3. An example of comparison can be provided in the solution folder, as a txt file or other file formats that is readable

I will work on this later, thanks

**Comment by kenghooi-teoh** [comment]
@KianYang-Lee pushed some code to get you started in the solutions folder, let's just work on enhancing e.g. example of explaining the question we asked in the instruction, loss monitoring etc.

**Comment by KianYang-Lee** [comment]
OK will work on the enhancement by the end of this week

**Comment by KianYang-Lee** [comment]
@kenghooi-teoh  feel free to review added tracking loss using UI

File: dl4j-cv-labs/src/main/java/ai/certifai/solution/classification/architectures/CompareCNN.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import org.deeplearning4j.optimize.listeners.ScoreIterationListener;
+import org.deeplearning4j.ui.api.UIServer;
+import org.deeplearning4j.ui.model.stats.StatsListener;
+import org.deeplearning4j.ui.model.storage.InMemoryStatsStorage;
+import org.nd4j.evaluation.classification.Evaluation;
+import org.nd4j.linalg.dataset.api.iterator.DataSetIterator;
+import org.nd4j.linalg.learning.config.Nesterovs;
+import org.slf4j.Logger;
+
+import javax.xml.crypto.Data;
```
Comment (kenghooi-teoh) [inline dl4j-cv-labs/src/main/java/ai/certifai/solution/classification/architectures/CompareCNN.java:39]: accidental import?

File: dl4j-cv-labs/src/main/java/ai/certifai/solution/classification/architectures/CompareCNN.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            graph.fit(trainIter, 1);
+            Evaluation eval = graph.evaluate(testIter);
+            System.out.println(name + "\n" + eval.stats());
+            log.info(name + "\n" + eval.stats());
+            System.out.println( name + " MODEL TRAINING ENDS" + "\n<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<\n\n\n" );
+            trainIter.reset();
+            testIter.reset();
+        }
+    }
+    private static void getListeners(ComputationGraph model){git
```
Comment (kenghooi-teoh) [inline dl4j-cv-labs/src/main/java/ai/certifai/solution/classification/architectures/CompareCNN.java:94]: the trailing 'git' needs to removed

**Comment by KianYang-Lee** [comment]
Oops fixed it
